### PR TITLE
Feature/initialize

### LINKED ISSUE #13: Player Stats
As a user, I should see my health, current health and name of the player character on the page. 
- [x] Sees current health of the player
- [x] Sees total health of the player
- [x] Sees the name of player

### PR DESCRIPTION
**Pull request by rubyr**
## Description

This change initializes a new react application (using `create-react-app`) and additionally inserts basic boilerplate code for global state management with react hooks

Fixes #13 
## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

There are no tests for these files as of yet. All changes have been tested manually.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+function App() {
+  const { initialize, hurt, heal } = usePlayerDispatch();
+  return (
+    <div className="App">
+      <UserStats />
+      <button onClick={() => initialize("sbeven")}>Init Player</button>
+      <button onClick={() => hurt(3)}>Hurt Player</button>
+      <button onClick={() => heal(3)}>Heal Player</button>
+    </div>
```
Comment (Corbinj22) [inline src/App/App.js:14]: Thanks for leaving these in @rubyr its cool seeing where the player is initialized.

File: src/playerState/index.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+} from "react";
+
+import { reducer, PlayerActions } from "./reducer";
+
+const initialContext = {
+  name: "",
+  hp: 0,
+  maxhp: 0,
+  attack: 0,
+  defense: 0,
```
Comment (Maxwell-Baird) [inline src/playerState/index.js:15]: So the User information will be stored on the front end?

File: src/playerState/reducer.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      return {
+        ...state,
+        hp: Math.min(state.maxhp, state.hp + action.hp),
+      };
+    case PlayerActions.HURT:
+      // todo: add in death code
+      return {
+        ...state,
+        hp: Math.max(0, state.hp - action.hp),
+      };
```
Comment (Maxwell-Baird) [inline src/playerState/reducer.js:33]: So this is where the player will update his information

File: src/playerState/index.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+const StateContext = createContext(initialContext);
+const DispatchContext = createContext(undefined);
+
+export const PlayerContextProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialContext);
```
Comment (Corbinj22) [inline src/playerState/index.js:22]: Ill have to look a bit more into the useReducer function. But love how this is set up! Great job.

**Pull request merged**